src/NHSD.BuyingCatalogue.SolutionLists.Persistence/Repositories/SolutionListRepository.cs

```diff
@@ -37,7 +37,17 @@ FROM    Solution
         /// Gets a list of <see cref="ISolutionListResult"/> objects.
         /// </summary>
         /// <returns>A list of <see cref="ISolutionListResult"/> objects.</returns>
-        public async Task<IEnumerable<ISolutionListResult>> ListAsync(bool foundationOnly, CancellationToken cancellationToken)
-            => await _dbConnector.QueryAsync<SolutionListResult>(foundationOnly ? sql + " AND COALESCE(FrameworkSolutions.IsFoundation, 0) = 1" : sql, cancellationToken).ConfigureAwait(false);
+        public async Task<IEnumerable<ISolutionListResult>> ListAsync(bool foundationOnly, string supplierId, CancellationToken cancellationToken)
+        {
+            var queryString = foundationOnly ? sql + " AND COALESCE(FrameworkSolutions.IsFoundation, 0) = 1" : sql;
+
+            if (supplierId != null)
+                queryString += $" AND Solution.SupplierId = '{supplierId}'";
+
+            return await _dbConnector
+                .QueryAsync<SolutionListResult>(
+                    queryString,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
I think the string interpolation may be susceptible to SQL injection. If you would prefer to keep the existing approach then we should update this to:

```c#
if (supplierId != null)
    queryString += " AND dbo.Solution.SupplierId = @supplierId;";

    return await _dbConnector.QueryAsync<SolutionListResult>(
        queryString,
        cancellationToken,
        supplierId);
```

Another option here would be to update `sql` to be

```sql
-- ...
WHERE dbo.Solution.PublishedStatusId = 3
  AND dbo.Solution.SupplierId = COALESCE(@supplierId, dbo.Solution.SupplierId)
```
which would avoid the need for the `if` and, as above, `QueryAsync` would become:

```c#
return await _dbConnector.QueryAsync<SolutionListResult>(
    queryString,
    cancellationToken,
    supplierId);
```

(I think the SQL would read better with table aliases but I don't think there's any point in tweaking the SQL formatting as part of this PR.)

Feel free to pick whichever approach you prefer that avoids the string interpolation.

Reply:
👍